schematic/utils/schema_utils.py

```diff
@@ -327,72 +372,74 @@ def parse_component_validation_rules(validation_rule_string: str) -> Dict:
     return validation_rules_dict
 
 
-def parse_single_set_validation_rules(validation_rule_string: str) -> list:
+def parse_single_set_validation_rules(validation_rule_string: str) -> list[str]:
     """Parse a single set of validation rules.
+
     Args:
-        validation_rule_string, str: validation rule provided by user.
+        validation_rule_string (str): validation rule provided by user.
+
     Returns:
-        list, the valiation rule string split by the rule delimiter
-    Raise:
-        ValueEror if the string contains a component name delimter in the beginning.
-            This would indicate that a user was trying to set component rules, but did so improperly.
+        list: the validation rule string split by the rule delimiter
     """
     # Try to catch an improperly formatted rule
     if COMPONENT_NAME_DELIMITER == validation_rule_string[0]:
         logger.error(
-            f"The provided validation rule {validation_rule_string}, looks to be formatted as a component "
-            f"based rule, but is missing the necessary formatting, "
-            f"please refer to the SchemaHub documentation for more details."
+            f"The provided validation rule {validation_rule_string}, looks to be formatted as a "
+            "component based rule, but is missing the necessary formatting, "
+            "please refer to the SchemaHub documentation for more details."
         )
 
     return validation_rule_string.split(RULE_DELIMITER)
 
 
-def parse_validation_rules(validation_rules: Union[list, dict]) -> Union[list, dict]:
+def parse_validation_rules(validation_rules: Any) -> Optional[Union[list, dict]]:
```

Comment:
In the PR for this function, for the typing, you indicated if we used union to restrict to list and dict, then we wouldn't have to add and additional else to look for other types. With it like this, its actually not being properly flagged when a str is passed. I think we should either add back the Union or add an additional test that would raise an exception for strings. Its passing the test right (`pytest tests/test_utils.py::TestSchemaUtils::test_parse_validation_rules[str_rule]` now because the asserts are in the try...I think we should update the test (perhaps in another PR) to move the asserts out of the try. Like this:

```try:
            parsed_validation_rules = parse_validation_rules(
                validation_rules=validation_rules
            )
        except:
            assert test_rule_name in ["str_rule", "duplicated_component"]

        if parsed_validation_rules:
            assert expected_parsed_rules == parsed_validation_rules
            assert test_rule_name not in ["str_rule", "duplicated_component"]
        else:
            test_rule_name in ["str_rule", "duplicated_component"]

Reply:
Yeah this is interesting, and I don't think there's one rigth answer. I guess it comes down to do you think its this functions job to handle strings (even if its to raise an error), or does it just handle lists/dicts and its up to the calling functions to make sure they are providing a list or dict? I would go with the latter, but we would need to make sure the functiosn that call it are doign so correctly.

Reply:
So this is the only places this function is called besides tests:

Reply:
Ok I tried the above and got a lot of errores in the tests

Reply:
I think I would rather explicitly restrict the inputs since we are expecting them to be only those, but anyone developing in the future might not know that, I would rather them be stopped.

Reply:
@mialy-defelice I changes the code where it's caled to:

```
elif rel_func == parse_validation_rules:
    rules = attr_relationships[csv_header]
    if  isinstance(rules, dict) or isinstance(rules, list):
        return parse_validation_rules(rules)
```
The function is now only called for lists and dicts

The funciton now looks like:

```
def parse_validation_rules(validation_rules: Union[list, dict]) -> Union[list, dict]:
    """Split multiple validation rules based on :: delimiter

    Args:
        validation_rules (Union[list, dict]): List or Dictionary of validation rules,
            if list:, contains a string validation rule
            if dict:, key is the component the rule (value) is applied to

    Returns:
        Union[list, dict]: Parsed validation rules, component rules are output
          as a dictionary, single sets are a list.
    """
    if isinstance(validation_rules, dict):
        # Rules pulled in as a dict can be used directly
        return validation_rules

    else:
        # If rules are already parsed from the JSONLD
        if len(validation_rules) > 1 and isinstance(validation_rules[-1], str):
            return validation_rules
        # Parse rules set for a subset of components/manifests
        if COMPONENT_RULES_DELIMITER in validation_rules[0]:
            return parse_component_validation_rules(
                validation_rule_string=validation_rules[0]
            )
        # Parse rules that are set across *all* components/manifests
        return parse_single_set_validation_rules(
            validation_rule_string=validation_rules[0]
        )
```
Now itexplicityl only takes lists or dicts.